samples/Calling/src/app/views/CallScreen.tsx

```diff
@@ -71,19 +64,60 @@ export const CallScreen = (props: CallScreenProps): JSX.Element => {
   );
 
   const credential = useMemo(
-    () => createAutoRefreshingCredential(toFlatCommunicationIdentifier(userId), token),
-    [token, userId]
+    () =>
+      isTeamsIdentityCall
+        ? new AzureCommunicationTokenCredential(token)
+        : createAutoRefreshingCredential(toFlatCommunicationIdentifier(userId), token),
+    [isTeamsIdentityCall, token, userId]
   );
 
-  /* @conditional-compile-remove(call-readiness) */
-  const options: CallCompositeOptions = useMemo(
-    () => ({
-      callReadinessOptedIn: callReadinessOptedIn,
-      onPermissionsTroubleshootingClick,
-      onNetworkingTroubleShootingClick
-    }),
-    [callReadinessOptedIn]
+  return isTeamsIdentityCall ? (
+    <TeamsCallScreen afterCreate={afterCreate} credential={credential} {...props} />
+  ) : (
+    <ACSCallScreen afterCreate={afterCreate} credential={credential} {...props} />
+  );
+};
+
+type CommonCallScreenProps = CallScreenProps & {
+  afterCreate?: <CallAdapterType extends CommonCallAdapter>(adapter: CallAdapterType) => Promise<CallAdapterType>;
+  credential: AzureCommunicationTokenCredential;
+};
+
+const TeamsCallScreen = (props: CommonCallScreenProps): JSX.Element => {
```

Comment:
there's so much Teams-logic vs ACS-logic, for the sample it would be nice if all ACS-call logic was in one file, and all Teams-Identity logic was in one file

Reply:
Maybe I can extract these 2 component into a new file

BTW, there is no big chunk of teams logic, the only different here is useAzureCommunicationTeamsCallAdapter vs useAzureCommunicationCallAdapter
But because of we can't call hooks conditionally, these 2 components becomes a must-have middle layer, sigh...

Reply:
I was moving code around for a while, then I noticed TeamsCallScreen and ACSCallScreen might be too small to stay in a file(especially without the context of callScreen)
But CallScreen file itself is too big to be a single file, then I realized I can extract the composite container part away and keep a nice balance of each file. (We definitely need some separate on the app level tbh)